fabric-client-gametest-api-v1/src/client/java/net/fabricmc/fabric/impl/client/gametest/ClientGameTestInputImpl.java

```diff
@@ -0,0 +1,333 @@
+		Preconditions.checkArgument(ticks > 0, "ticks must be positive");
+
+		KeyBinding keyBinding = context.computeOnClient(client -> keyBindingGetter.apply(client.options));
+		holdKeyFor(keyBinding, ticks);
+	}
+
+	@Override
+	public void holdKeyFor(InputUtil.Key key, int ticks) {
+		ThreadingImpl.checkOnGametestThread("holdKeyFor");
+		Preconditions.checkNotNull(key, "key");
+		Preconditions.checkArgument(ticks > 0, "ticks must be positive");
+
+		holdKey(key);
+		context.waitTicks(ticks);
+		releaseKey(key);
+	}
+
+	@Override
+	public void holdKeyFor(int keyCode, int ticks) {
+		ThreadingImpl.checkOnGametestThread("holdKeyFor");
+		Preconditions.checkArgument(ticks > 0, "ticks must be positive");
+
+		holdKeyFor(InputUtil.Type.KEYSYM.createFromCode(keyCode), ticks);
+	}
+
+	@Override
+	public void holdMouseFor(int button, int ticks) {
+		ThreadingImpl.checkOnGametestThread("holdMouseFor");
+		Preconditions.checkArgument(ticks > 0, "ticks must be positive");
+
+		holdKeyFor(InputUtil.Type.MOUSE.createFromCode(button), ticks);
+	}
+
+	@Override
+	public void typeChar(int codePoint) {
+		ThreadingImpl.checkOnGametestThread("typeChar");
+
+		context.runOnClient(client -> ((KeyboardAccessor) client.keyboard).invokeOnChar(client.getWindow().getHandle(), codePoint, 0));
+	}
+
+	@Override
+	public void typeChars(String chars) {
+		ThreadingImpl.checkOnGametestThread("typeChars");
+
+		context.runOnClient(client -> {
+			chars.chars().forEach(codePoint -> {
+				((KeyboardAccessor) client.keyboard).invokeOnChar(client.getWindow().getHandle(), codePoint, 0);
+			});
+		});
+	}
+
+	@Override
+	public void scroll(double amount) {
+		ThreadingImpl.checkOnGametestThread("scroll");
+
+		scroll(0, amount);
+	}
+
+	@Override
+	public void scroll(double xAmount, double yAmount) {
+		ThreadingImpl.checkOnGametestThread("scroll");
+
+		context.runOnClient(client -> ((MouseAccessor) client.mouse).invokeOnMouseScroll(client.getWindow().getHandle(), xAmount, yAmount));
+	}
+
+	@Override
+	public void setCursorPos(double x, double y) {
+		ThreadingImpl.checkOnGametestThread("setCursorPos");
+
+		context.runOnClient(client -> ((MouseAccessor) client.mouse).invokeOnCursorPos(client.getWindow().getHandle(), x, y));
+	}
+
+	@Override
+	public void moveCursor(double deltaX, double deltaY) {
+		ThreadingImpl.checkOnGametestThread("moveCursor");
+
+		context.runOnClient(client -> {
+			double newX = client.mouse.getX() + deltaX;
+			double newY = client.mouse.getY() + deltaY;
+			((MouseAccessor) client.mouse).invokeOnCursorPos(client.getWindow().getHandle(), newX, newY);
+		});
+	}
+
+	private static InputUtil.Key getBoundKey(KeyBinding keyBinding, String action) {
+		InputUtil.Key boundKey = ((KeyBindingAccessor) keyBinding).getBoundKey();
+
+		if (boundKey == InputUtil.UNKNOWN_KEY) {
+			throw new AssertionError("Cannot %s binding '%s' because it isn'keyBinding bound to a key".formatted(action, keyBinding.getTranslationKey()));
```

Comment:
Seems to be a typo: `isn'keyBinding`

Reply:
Oops, I probably copy pasted `keyBinding` and somehow replaced the `t` with it